### ISSUE TITLE
New Component: select with icon, title, snippet selection

### ISSUE THREAD

**Issue by AdelheidF** [Esri/calcite-design-system#4334]
### Description

closed select

![image](https://user-images.githubusercontent.com/44481292/160914113-cb2b7dd7-d14f-496f-8ec0-f690192eb9f8.png)

different versions of existing selection lists in the Map Viewer.

![image](https://user-images.githubusercontent.com/44481292/160914066-3041f9d5-25c7-4ac2-ba75-b590d8f895e9.png)![image](https://user-images.githubusercontent.com/44481292/160914311-6ca124c3-6997-4133-8f45-a4eb1e2b0335.png)

![image](https://user-images.githubusercontent.com/44481292/160914789-c3a5b75f-6265-434a-85eb-d056d2341f4d.png)![image](https://user-images.githubusercontent.com/44481292/160914862-4908f35e-1e77-4b7f-a482-b17361476267.png)

![image](https://user-images.githubusercontent.com/44481292/160919756-a2db0535-3ca7-4913-95c0-2b55ab3b154b.png)

The top 2 versions where the closed button displays the entire info instead of just the title is the way we want to have it used in the future.

### User Stories

Above pictures are all from the Map Viewer styles component.

### Acceptance Criteria

selection list should support at least a starting icon, a title and a snippet

**monday.com sync:** #9363907700

**Comment by macandcheese** [comment]
This could perhaps be a configuration of single-select combobox where the summary and slotted content is displayed at idle in addition to in the dropdown.

**Comment by macandcheese** [comment]
Dashboard's "category selector" widget is a good example of this kind of lockup too.

**Comment by macandcheese** [comment]
+1 - Component Sample on documentation site has a need for this as well.

**Comment by Anastasiia-Boleiko** [comment]
Also interested in this component. 
Example in ArcGIS Urban:
<img width="260" alt="Screenshot 2023-03-13 at 09 06 08" src="https://user-images.githubusercontent.com/119309703/225665090-9b69d506-0be1-425b-8294-e051efbe9092.png">

**Comment by stefanobettinelli** [comment]
Interesting a similar component.
Example of what we will be implementing with a temporary custom solution in Scene Viewer:
<img width="381" alt="Screenshot 2023-03-17 at 10 29 28" src="https://user-images.githubusercontent.com/2855964/225935718-d5c32d21-988b-4fab-aa21-69ee1050e23d.png">

**Comment by macandcheese** [comment]
This should also have an "unselected" entry point that allows filtering - I think the Combobox could support this with the internal stack.

**Comment by macandcheese** [comment]
This could also be a riff on List, where a single-selection List has a "collapsed view" of the selected item, or a placeholder. This Codepen (roughly) implements this with two separate Lists: https://codepen.io/mac_and_cheese/pen/YzdzyyY?editors=1000

With an already populated selection (this is slotting Avatar, it could slot anything needed from the selected list item - image, icon, etc.)
closed:
<img width="700" alt="Screenshot 2023-08-17 at 1 10 07 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/20ed0664-defc-4a1f-a3a6-8c5e34ae7f97">

open:
<img width="700" alt="Screenshot 2023-08-17 at 1 10 14 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/503312fe-3cb5-4d88-b739-ebc87471d75a">



With a "placeholder" invoking list item (again, can be customized as needed):
closed:
<img width="700" alt="Screenshot 2023-08-17 at 1 13 25 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/b8d72247-6c3b-4b1b-810c-1b4ae6d2cfbb">
open:
<img width="700" alt="Screenshot 2023-08-17 at 1 13 35 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/ee6a3577-8ec3-4e0d-aa92-8476edf474ea">


Some other ideas for the display of the invoking List Item - of course customizable per use case:

<img width="700" alt="Screenshot 2023-08-17 at 1 07 46 PM" src="https://github.com/Esri/calcite-design-system/assets/4733155/a8edb2fb-9482-439f-81e6-1ddceef20f59">

**Comment by janett-baresel** [comment]
This has come up again in Scene Viewer for our elevation mode drop-down:

![image (8)](https://github.com/Esri/calcite-design-system/assets/11048042/ce385374-07c1-441a-b68c-387dc2c2989d)

Any way to upvote this? @macandcheese

**Comment by macandcheese** [comment]
Thoughts on closing this cc @Esri/calcite-designers. These UX can be achieved in a few ways already...

**Comment by ashetland** [comment]
Yep. If there's a particular piece we don't currently cover, a new enhancement request to an existing component should be opened.
